driver/src/util.rs

```diff
@@ -1,36 +1,54 @@
 use crate::contract::SnappContract;
 use crate::error::DriverError;
+use crate::error::ErrorKind;
 
-use web3::types::U256;
+use web3::types::{H256, U256};
 
 pub fn find_first_unapplied_slot(
-    upper_bound: U256, 
-    has_slot_been_applied: Box<&Fn(U256) -> Result<bool, DriverError>>
-) -> Result<U256, DriverError>
-{
+    upper_bound: U256,
+    has_slot_been_applied: Box<&Fn(U256) -> Result<bool, DriverError>>,
+) -> Result<U256, DriverError> {
     if upper_bound == U256::max_value() {
         return Ok(U256::zero());
     }
     let mut slot = upper_bound + 1;
     while slot != U256::zero() {
         if has_slot_been_applied(slot - 1)? {
-            return Ok(slot)
+            return Ok(slot);
         }
         slot = slot - 1;
     }
     Ok(U256::zero())
 }
 
+pub fn hash_consistency_check(
+    hash_calculated: H256,
+    hash_from_contract: H256,
+    flux_type: &str,
+) -> Result<(), DriverError> {
+    if hash_calculated != hash_from_contract {
+        return Err(DriverError::new(
+            &format!(
```

Comment:
Could we merge the logger first and then change this to a log error... Or does it matter here since an Err is being returned anyway?